apps/client/src/features/rundown/Rundown.tsx

```diff
@@ -110,10 +118,25 @@ export default function Rundown({ data }: RundownProps) {
         newCursor =
           (direction === 'up' ? getLastNormal(rundown, order)?.id : getFirstNormal(rundown, order)?.id) ?? null;
         newIndex = direction === 'up' ? order.length : 0;
+
+        if (jumpBlock && newCursor) {
```

Comment:
Question: I think the logic here is becoming too nested and hard to follow.
We should look for ways to improve this.

With the current structure, we assign the values at the end, meaning we need to mentally hold the values and execution of the entire function to understand what it does.
Itw also seems that there is little overlap

I would suggest over the blocks and event logic

- add early returns for logic blocks
- split the block logic to a new function (may need to rename the functions to `selectBlock` and `selectEvent`)